docs/docs/concepts/05-building-blocks/01-signals.md

```diff
@@ -16,15 +16,16 @@ Often, the message topic and message type is hardcoded within the implementation
 difficult to rearrange the network topology of ROS nodes without modifying and recompiling the node implementation
 itself.
 
+<!-- TODO: we now support all message types and not just standardized -->
 Signals then are ROS 2 publishers and subscribers with dynamically assigned topics and standardized message types.
 This makes it easy to reconfigure the signal connections between different components and controllers in the application
 graph without modifying or recompiling any source code. By using standard message types, the signal compatibility
-between components and controllers is also simplified.
+between components and controllers is also simplified. When components and controllers are connected by a signal in an
+application graph, the application interpreter will try to assert that the signals have a matching type.
 
 Additionally, ROS 2 messages are data packets, not data objects. Parsing data from a message, manipulating it and
-writing it back into a message can involve a fair amount of boilerplate code.
-
-When developing an AICA component, signals are automatically converted to and from the corresponding data object.
+writing it back into a message can involve a fair amount of boilerplate code. When developing an AICA component, signals
+are usually automatically converted to and from the corresponding data object.
 
 ## Basic signal types
```

Comment:
Also in a follow up PR, I want to provide a table that explains what these message types are mapped to, e.g. `double` -> `Float64`

Reply:
One of my concerns reading this is that we say:

> Additionally, ROS 2 messages are data packets, not data objects. Parsing data from a message, manipulating it and writing it back into a message can involve a fair amount of boilerplate code. When developing an AICA component, signals are usually automatically converted to and from the corresponding data object.

but I don't find it accurate. ROS messages are structs after all. This also feels like we are deviating from ROS language a bit (which we eventually have to but we should make a clear connection first).

I would personally prefer/suggest something more along the lines of

> ROS 2 messages are lean data structures with no additional functionality other than storing data. Therefore, parsing data from a message, manipulating it and writing it back into a message can involve a fair amount of boilerplate code. Conversely, the AICA System will typically convert these message structures to and from objects that encapsulate a lot more functionality for manipulating, transforming, and in general operating on their contents.